### PR TITLE
treewide: Rename nixfmt to nixfmt-classic

### PR DESCRIPTION
**Pull request by dasJ**
## Description of changes

Opens up the attribute for future use :)

cc @NixOS/nix-formatting 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Is it OK to change the `override`  interface?

**Comment by infinisil** [comment]
Let's discuss this a bit more in the formatting team before deciding what to do with the nixfmt attribute :)

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -845,6 +845,7 @@ mapAliases ({
   nix_2_4 = nixVersions.nix_2_4;
   nix_2_5 = nixVersions.nix_2_5;
   nix_2_6 = nixVersions.nix_2_6;
+  nixfmt = lib.warn "nixfmt was renamed to nixfmt-classic. The nixfmt attribute will be used for the new RFC 166-style formatter in the future." nixfmt-classic; # Added 2024-03-31
```
Comment (infinisil) [inline pkgs/top-level/aliases.nix:848]:
```suggestion
  nixfmt = lib.warn "nixfmt was renamed to nixfmt-classic. The nixfmt attribute may be used for the new RFC 166-style formatter in the future, which is currently available as nixfmt-rfc-style" nixfmt-classic; # Added 2024-03-31
```

**Review by infinisil — APPROVED** [review]
(reviewed in the formatting team meeting)

Other than the suggestion, looking good!

<details><summary>Detailed discussion</summary>

  - Main motivation: Avoid people being confusing when `nixfmt` suddenly changes completely
  - Deprecate the `nixfmt` attribute for now, rename to `nixfmt-classic` seems reasonable
  - Can continue using `nixfmt-rfc-style` for now, no hurry to reuse the `nixfmt` attribute

</details>

**Comment by infinisil** [comment]
> Is it OK to change the `override` interface?

Oh, we glossed over that. I personally am a big proponent of keeping the `.override` interface stable. @dasJ would you be okay with reverting and doing it with `nixfmt = nixfmt-classic;` instead to fix that?

**Comment by dasJ** [comment]
Without the warning? Why not

**Comment by infinisil** [comment]
@dasJ Oh no, I mean instead of changing each `default.nix` file, keep those the same, and just do this in `all-packages.nix`:

```nix
  foo = callPackage ./foo.nix {
    nixfmt = nixfmt-classic;
  };
```

**Pull request merged**